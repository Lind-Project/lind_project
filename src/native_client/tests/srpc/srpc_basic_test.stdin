rpc bool b(0) * b(0)
rpc bool b(1) * b(0)
rpc double d(1.0) * d(0)
rpc nan * d(0)
rpc int i(1) * i(0)
rpc long l(1) * l(0)
rpc string s("hello") * i(0)
rpc char_array C(5,hello) * C(5)
rpc char_array C(12,\110\145\154\154\157,\x77\x6f\x72\x6C\x64\X2E) * C(12)
rpc char_array C(9,A\b\f\n\t\"\"\\\x7F) * C(9)
rpc char_array R(7,1:0x88,2:0xaa99,4:0xeeddccbb) * C(7)
rpc double_array D(5,3.1,1.4,4.1,1.5,5.9) * D(5)
rpc int_array I(5,1,2,3,4,5) * I(5)
rpc long_array L(5,1,2,3,4,5) * L(5)
rpc null_method *
rpc stringret i(0) * s("")
rpc handle h(stderr) *
rpc handleret * h(0)
rpc invalid_handle h(invalid) *
rpc invalid_handle_ret * h(0)
