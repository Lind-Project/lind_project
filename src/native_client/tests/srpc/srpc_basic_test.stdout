rpc call initiated bool:b:b
input 0:  b(0)
rpc call complete bool:b:b
output 0:  b(1)

rpc call initiated bool:b:b
input 0:  b(1)
rpc call complete bool:b:b
output 0:  b(0)

rpc call initiated double:d:d
input 0:  d(1)
rpc call complete double:d:d
output 0:  d(-1)

rpc call initiated nan::d
rpc call complete nan::d
output 0:  d(NaN)

rpc call initiated int:i:i
input 0:  i(1)
rpc call complete int:i:i
output 0:  i(-1)

rpc call initiated long:l:l
input 0:  l(1)
rpc call complete long:l:l
output 0:  l(-1)

rpc call initiated string:s:i
input 0:  s("hello")
rpc call complete string:s:i
output 0:  i(5)

rpc call initiated char_array:C:C
input 0:  C(5,hello)
rpc call complete char_array:C:C
output 0:  C(5,olleh)

rpc call initiated char_array:C:C
input 0:  C(12,Hello,world.)
rpc call complete char_array:C:C
output 0:  C(12,.dlrow,olleH)

rpc call initiated char_array:C:C
input 0:  C(9,A\b\f\n\t\"\"\\\x7f)
rpc call complete char_array:C:C
output 0:  C(9,\x7f\\\"\"\t\n\f\bA)

rpc call initiated char_array:C:C
input 0:  C(7,\x88\x99\xaa\xbb\xcc\xdd\xee)
rpc call complete char_array:C:C
output 0:  C(7,\xee\xdd\xcc\xbb\xaa\x99\x88)

rpc call initiated double_array:D:D
input 0:  D(5,3.1,1.4,4.1,1.5,5.9)
rpc call complete double_array:D:D
output 0:  D(5,5.9,1.5,4.1,1.4,3.1)

rpc call initiated int_array:I:I
input 0:  I(5,1,2,3,4,5)
rpc call complete int_array:I:I
output 0:  I(5,5,4,3,2,1)

rpc call initiated long_array:L:L
input 0:  L(5,1,2,3,4,5)
rpc call complete long_array:L:L
output 0:  L(5,5,4,3,2,1)

rpc call initiated null_method::
rpc call complete null_method::

rpc call initiated stringret:i:s
input 0:  i(0)
rpc call complete stringret:i:s
output 0:  s("Ich weiss nicht, was soll es bedeutenDass ich so traurig bin,Ein Maerchen aus uralten Zeiten,Das kommt mir nicht aus dem Sinn.Die Luft ist kuehl und es dunkelt,Und ruhig fliesst der Rhein;Der Gipfel des Berges funkelt,Im Abendsonnenschein.Die schoenste Jungfrau sitzetDort oben wunderbar,Ihr gold'nes Geschmeide blitzet,Sie kaemmt ihr goldenes Haar,Sie kaemmt es mit goldenem Kamme,Und singt ein Lied dabei;Das hat eine wundersame,Gewalt'ge Melodei.")

rpc call initiated handle:h:
input 0:  h(stderr)
rpc call complete handle:h:

rpc call initiated handleret::h
rpc call complete handleret::h
output 0:  h(imported_0)

rpc call initiated invalid_handle:h:
input 0:  h(invalid)
rpc call complete invalid_handle:h:

rpc call initiated invalid_handle_ret::h
rpc call complete invalid_handle_ret::h
output 0:  h(invalid)

